@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+const SEGMENT_STRIDE = 6;
+// Stride between the start and end vertex indices of a segment. Vertices are duplicated so that
+// each copy is extruded in opposite directions in the vertex shader.
+const VERTEX_STRIDE = 2;
+
+function isSolidLineMaterial(material: THREE.Material | THREE.Material[]): boolean {
+    return Array.isArray(material)
+        ? material.every(mat => mat instanceof SolidLineMaterial)
+        : material instanceof SolidLineMaterial;
+}
+
+/**
+ * Computes the bounding sphere of the part of a given geometry corresponding to a feature.
+ * @param geometry The geometry containing the feature.
+ * @param featureBeginIndex The index where the feature starts in the geometry's indices attribute.
+ * @param featureEndIndex The index where the feature end in the geometry's indices attribute.
+ * @returns The feature bounding sphere.
+ */
+function computeFeatureBoundingSphere(
+    geometry: THREE.BufferGeometry | DisplacedBufferGeometry,
+    featureBeginIndex: number,
+    featureEndIndex: number
+): THREE.Sphere {
+    let displacementRange: DisplacementRange | undefined;
+
+    if (geometry instanceof DisplacedBufferGeometry) {
+        displacementRange = geometry.displacementRange;
+        geometry = geometry.originalGeometry;
+    }
+
+    const attributes = geometry.attributes;
+    const pos = attributes.position as THREE.BufferAttribute;
+    const indices = geometry.index!.array;
+    const sphere = new THREE.Sphere();
+    const bbox = tmpBox.makeEmpty();
+    const vertex = tmpV1;
+
+    // First compute the bounding box for all line segments.
+    for (let i = featureBeginIndex; i < featureEndIndex; i += SEGMENT_STRIDE) {
+        bbox.expandByPoint(vertex.fromBufferAttribute(pos, indices[i]));
+        bbox.expandByPoint(vertex.fromBufferAttribute(pos, indices[i + VERTEX_STRIDE]));
+    }
+
+    const normal = tmpV2;
+    if (displacementRange) {
+        // If geometry is displaced, expand the bounding box to cover the whole displacement range,
+        // and return the sphere bounding the box. This is a coarse estimation, but avoids having
+        // to displace all vertices.
+        return expandBoxByDisplacementRange(
+            bbox,
+            displacementRange,
+            normal.fromBufferAttribute(geometry.attributes.normal as THREE.BufferAttribute, 0)
+        ).getBoundingSphere(sphere);
+    }
+
+    // If geometry is not displaced, the bounding sphere is placed in the center of the computed
+    // bounding box and expanded until it contains all vertices.
+    const center = bbox.getCenter(sphere.center);
+    let maxRSq = 0;
+    for (let i = featureBeginIndex; i < featureEndIndex; i += SEGMENT_STRIDE) {
+        maxRSq = Math.max(
+            maxRSq,
+            center.distanceToSquared(vertex.fromBufferAttribute(pos, indices[i])),
+            center.distanceToSquared(vertex.fromBufferAttribute(pos, indices[i + VERTEX_STRIDE]))
+        );
+    }
+    sphere.radius = Math.sqrt(maxRSq);
```

Comment:
I don't understand what you want to do here then. Do you say we should just take the sphere where the cube is inscribed instead of trying to find a smaller one?

Reply:
Ok, i get it now. I'll remove the radius fitting step and use the box extent instead. 

> What I am trying to say that the approach used is neither accurate, nor fast. I simply see no point of calculating it, especially taking into account that in the same PR you are using `box.getBoundingSphere(sphere)` - very very rough approximation.

That very rough approximation is done when geometry is displaced. In that case you can't find a tight bounding volume unless you actually displace the whole geometry, vertex by vertex, that's what i try to avoid.